website/versioned_docs/version-0.1.0/Connection/5.mongodb_connection.md

```diff
@@ -0,0 +1,137 @@
+
+---
+id: mongodb_connection_registry
+title: MongoDB Connection Registry
+sidebar_label: MongoDB Connection Registry
+---
+
+
+`MongoDBConnection` provide the `mongo_client` on every call to the `get` method. And will `close` the connection when exiting the `"with"` statement.
+
+## Example
+##### config.yaml
+
+```yaml
+core_lib:
+  mongodb:
+    _target_: core_lib.connection.mongodb_connection_registry.MongoDBConnectionRegistry
+    config:
+      url:
+        protocol: mongodb+srv
+        host: server.example.com
+        port: 27017
+        username: test
+        password: test123
+```
+
+##### mongodb_connection.py
+
+```python
+from pymongo import MongoClient
+from core_lib.connection.connection import Connection
+
+
+class MongoDBConnection(Connection):
+    def __init__(self, mongo_client: MongoClient):
+        self.mongo_client = mongo_client
+
+    def __enter__(self) -> MongoClient:
+        return self.mongo_client
+
+    def __exit__(self, exec_type, exec_value, traceback):
+        pass
+```
+
+#### mongodb_connection_registry.py
+
+```python
+from omegaconf import DictConfig
+import pymongo
+from core_lib.connection.connection_registry import ConnectionRegistry
+from core_lib.connection.mongodb_connection import MongoDBConnection
+from core_lib.data_layers.data.data_helpers import build_url
+
+
+class MongoDBConnectionRegistry(ConnectionRegistry):
+
+    def __init__(self, config: DictConfig):
+        self._mongo_client = pymongo.MongoClient(build_url(**config.url))
+
+    @property
+    def client(self) -> pymongo.MongoClient:
+        return self._mongo_client
+
+    def get(self, *args, **kwargs):
+        return MongoDBConnection(self._mongo_client)
+```
+
+# MongoDBConnection
+
+*core_lib.connection.mongodb_connection.MongoDBConnection* [[source]](https://github.com/shay-te/core-lib/blob/master/core_lib/connection/mongodb_connection.py#L6)
+
+This class gets the `mongo_client` on initialization.
+
+```python
+class MongoDBConnection(Connection):
+    def __init__(self, mongo_client: pymongo.MongoClient):
+        self.mongo_client = mongo_client
+```
+**Arguments**
+
+- **`mongo_client`** *`(MongoClient)`*: The `mongo_client` provided from  `MongoDBConnectionRegistry`.
+
+

```

Comment:
see how it looks in the live docs, add another enter

Reply:
It looks the same even after adding another enter